ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -188,4 +197,344 @@ const BaseLine = (props) => {
   )
 }
 
+const LineDms = (props) => {
+  const {
+    geometryKey,
+    dmsPointArray,
+    setState,
+    unitKey,
+    setBufferState,
+    widthKey,
+  } = props
+  const [baseLineError, setBaseLineError] = useState(initialErrorState)
+  const [bufferError, setBufferError] = useState(initialErrorState)
+
+  useEffect(() => {
+    if (props.drawing) {
+      setBaseLineError(initialErrorState)
+    }
+    if (dmsPointArray) {
+      setBaseLineError(validateDmsLineOrPoly(dmsPointArray, geometryKey))
+    }
+  }, [props.polygon, props.line, dmsPointArray])
+
+  return (
+    <div>
+      <div className="input-location">
+        {dmsPointArray &&
+          dmsPointArray.map((point, index) => {
```

Comment:
✏️ I believe you can use optional chaining here `dmsPointArray?.map(() => do stuff)`.

Reply:
Oh you are right, I blanked on the syntax for this, good call 👍